core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -281,14 +282,8 @@ class RemoteIndexCacheTest {
     // no expired entries yet
     assertEquals(0, cache.expiredIndexes.size, "expiredIndex queue should be zero at start of test")
 
-    // invalidate the cache. it should async mark the entry for removal
-    cache.internalCache.invalidate(internalIndexKey)
-
-    // wait until entry is marked for deletion
-    TestUtils.waitUntilTrue(() => cacheEntry.isMarkedForCleanup,
-      "Failed to mark cache entry for cleanup after invalidation")
-    TestUtils.waitUntilTrue(() => cacheEntry.isCleanStarted,
-      "Failed to cleanup cache entry after invalidation")
+    // call remove function to mark the entry for removal
+    cache.remove(internalIndexKey)
 
     // first it will be marked for cleanup, second time markForCleanup is called when cleanup() is called
     verify(cacheEntry, times(2)).markForCleanup()
```

Comment:
In L291, we want to verify `cleanup` is invoked, but after my suggestion to remove `cleanup` in `internalCache.asMap().computeIfPresent`, now, we need to add this before verify it.
```
    TestUtils.waitUntilTrue(() => cacheEntry.isCleanStarted,
      "Failed to cleanup cache entry after invalidation")
```

Reply:
I removed `cleanup` call in `internalCache.asMap().computeIfPresent` and added wait as well but cleanup is not getting called.